server/v2/cometbft/server.go

```diff
@@ -155,15 +162,21 @@ func (s *CometBFTServer[T]) Stop(_ context.Context) error {
 	return nil
 }
 
-func (s *CometBFTServer[T]) Config() (any, *viper.Viper) {
-	v := viper.New()
-	v.SetConfigFile("???") // TODO: where do we set this
-	v.SetConfigName("config")
-	v.SetConfigType("toml")
-	if err := v.ReadInConfig(); err != nil {
-		panic(fmt.Errorf("failed to read cometbft config file: %w", err))
+func (s *CometBFTServer[T]) Config() any {
```

Comment:
We shouldn't need this, neither the one afterwards.
We can let cometbft handle the config directly in New or Start. Additionally, comet should use `cmtcfg.WriteConfigFile` to write its config properly.